ml_genn/layers/avepool2d_conv2d_synapses.py

```diff
@@ -48,31 +48,35 @@
     const int poolInChan = $(id_pre) % pool_ic;
 
     // Calculate corresponding pool output
-    const int poolOutRow = poolInRow  / pool_sh;
+    const int poolOutRow = poolInRow / pool_sh;
     const int poolStrideRow = poolOutRow * pool_sh;
     const int poolOutCol = poolInCol / pool_sw;
     const int poolStrideCol = poolOutCol * pool_sw;
 
     if ((poolInRow < (poolStrideRow + pool_kh)) && (poolInCol < (poolStrideCol + pool_kw))) {
-        // Calculate range of output rows and columns which this pool output connects to
-        const int minOutRow = min(conv_oh, max(0, 1 + ((poolOutRow + conv_padh - conv_kh) / conv_sh)));
-        const int maxOutRow = min(conv_oh, max(0, 1 + ((poolOutRow + conv_padh) / conv_sh)));
-        const int minOutCol = min(conv_ow, max(0, 1 + ((poolOutCol + conv_padw - conv_kw) / conv_sw)));
-        const int maxOutCol = min(conv_ow, max(0, 1 + ((poolOutCol + conv_padw) / conv_sw)));
-
-        // Loop through output rows, columns and channels
-        for(int convOutRow = minOutRow; convOutRow < maxOutRow; convOutRow++) {
-            const int strideRow = (convOutRow * conv_sh) - conv_padh;
-            const int kernRow = poolOutRow - strideRow;
-            for(int convOutCol = minOutCol; convOutCol < maxOutCol; convOutCol++) {
-                const int strideCol = (convOutCol * conv_sw) - conv_padw;
-                const int kernCol = poolOutCol - strideCol;
-                for(int outChan = 0; outChan < conv_oc; outChan++) {
-                    // Calculate postsynaptic index and add synapse
-                    const int idPost = ((convOutRow * conv_ow * conv_oc) +
-                                        (convOutCol * conv_oc) +
-                                        outChan);
-                    $(addSynapse, idPost, kernRow, kernCol, poolInChan, outChan);
+        const int conv_ih = $(conv_ih), conv_iw = $(conv_iw), conv_ic = $(conv_ic);
+
+        if ((poolOutRow < conv_ih) && (poolOutCol < conv_iw)) {
+            // Calculate range of output rows and columns which this pool output connects to
+            const int minOutRow = min(conv_oh, max(0, 1 + ((poolOutRow + conv_padh - conv_kh) / conv_sh)));
```

Comment:
These min calculations should be done like:
```c++
min((int)$(conv_oh), max(0, 1 + (int)floor((inRow + $(conv_padh) - $(conv_kh)) / $(conv_sh))))
```
to deal with the rounding issue with stride described in https://github.com/genn-team/genn/pull/489 (again, sorry for not doing this)